R/R.sublime-syntax

```diff
@@ -187,123 +235,242 @@ contexts:
-          pop: true
+          pop: 1
+
+  single-quoted-string-escapes:
+    - meta_include_prototype: false
+    - match: (\\)\'
+      scope: constant.character.escape.r
+      captures:
+        1: punctuation.definition.string.escape.r
+    - include: string-escapes
+
+  double-quoted-string-escapes:
+    - meta_include_prototype: false
+    - match: (\\)\"
+      scope: constant.character.escape.r
+      captures:
+        1: punctuation.definition.string.escape.r
+    - include: string-escapes
+
+  string-escapes:
+    - meta_include_prototype: false
+    - include: invalid-string-escapes
+    - include: valid-string-escapes
+
+  invalid-string-escapes:
+    - meta_include_prototype: false
+    - match: '(\\)0'                                        # null character
+      scope: invalid.illegal.constant-character-escape.r
+
+  valid-string-escapes:
+    - meta_include_prototype: false
+    - match: '(\\)\\'                                       # backslash
+      scope: constant.character.escape.r
+      captures:
+        1: punctuation.definition.string.escape.r
+    - match: '(\\)n'                                        # newline
+      scope: constant.character.escape.r
+      captures:
+        1: punctuation.definition.string.escape.r
+    - match: '(\\)r'                                        # carriage return
+      scope: constant.character.escape.r
+      captures:
+        1: punctuation.definition.string.escape.r
+    - match: '(\\)t'                                        # horizontal tab
+      scope: constant.character.escape.r
+      captures:
+        1: punctuation.definition.string.escape.r
+    - match: '(\\)b'                                        # backspace
+      scope: constant.character.escape.r
+      captures:
+        1: punctuation.definition.string.escape.r
+    - match: '(\\)a'                                        # bell
+      scope: constant.character.escape.r
+      captures:
+        1: punctuation.definition.string.escape.r
+    - match: '(\\)f'                                        # form feed
+      scope: constant.character.escape.r
+      captures:
+        1: punctuation.definition.string.escape.r
+    - match: '(\\)v'                                        # vertical tab
+      scope: constant.character.escape.r
+      captures:
+        1: punctuation.definition.string.escape.r
+    - match: '(\\)[0-7]{1,3}'
+      scope: constant.character.escape.r
+      captures:
+        1: punctuation.definition.string.escape.r
+    - match: '(\\)x\h{1,2}'
+      scope: constant.character.escape.r
+      captures:
+        1: punctuation.definition.string.escape.r
+    - match: '(\\)u\{?\h{1,4}\}?'
+      scope: constant.character.escape.r
+      captures:
+        1: punctuation.definition.string.escape.r
+    - match: '(\\)U\{?\h{1,8}\}?'
+      scope: constant.character.escape.r
+      captures:
+        1: punctuation.definition.string.escape.r
```

Comment:
1. Few syntaxes certainly scope `\\` as `punctuation.definition.escape`. As most of them consist of only 2 characters, it adds up more noise than use. Hence I'd rather vote to remove existing punctuation scope from backslash escapes instead of adding more.
2. It is not useful to add a single pattern for each individual escape sequence. I'd rather suggest to group them. Good examples (also for scoping) is Python.

   https://github.com/sublimehq/Packages/blob/5fa6fb77a536dabaf5abb5e6c6bc309d03708d75/Python/Python.sublime-syntax#L4039-L4055